tests/RobotFramework/tests/mqtt/custom_sub_topics_tedge-mapper-az.robot

```diff
@@ -11,20 +11,85 @@ Test Tags           theme:mqtt    theme:az
 
 
 *** Test Cases ***
-Publish measurements to subscribed topic
-    Execute Command    tedge mqtt pub tedge/measurements/child '{"temperature": 10}'
-    Should Have MQTT Messages    az/messages/events/#
-
-Publish measurements to unsubscribed topic
-    Execute Command    tedge mqtt pub tedge/measurements '{"temperature": 10}'
+Publish measurements to unsubscribed topic   
+    Execute Command    sudo tedge config unset az.topics
+    Execute Command    sudo systemctl restart tedge-mapper-az.service
     Sleep    5s    reason=If a message is not published in 5s, it will never be published.
-    Should Have MQTT Messages    az/messages/events/#    minimum=0    maximum=0
+    Should Have MQTT Messages    az/messages/events/device:main/m/    minimum=0    maximum=0
+
+Publish measurements to te measurement topic
+    ${timestamp}=        Get Unix Timestamp
+    Execute Command    sudo tedge config set az.topics "te/+/+/+/+/m/+"
+    Execute Command    sudo systemctl restart tedge-mapper-az.service
+    ThinEdgeIO.Service Health Status Should Be Up    tedge-mapper-az
+    Execute Command    tedge mqtt pub te/device/main///m/ '{"temperature": 10}'
+    Should Have MQTT Messages    az/messages/events/device:main/m/   message_contains="temperature"    date_from=${timestamp}   minimum=1    maximum=1
+
+Publish measurements to te measurement topic with measurement type
+    ${timestamp}=        Get Unix Timestamp
+    Execute Command    sudo tedge config set az.topics "te/+/+/+/+/m/+"
+    Execute Command    sudo systemctl restart tedge-mapper-az.service    
+    ThinEdgeIO.Service Health Status Should Be Up    tedge-mapper-az
+    Execute Command    tedge mqtt pub te/device/main///m/test-type '{"temperature": 10}'    Should Have MQTT Messages    az/messages/events/device:main/m/test-type   message_contains="temperature"    date_from=${timestamp}   minimum=1    maximum=1
```

Comment:
`Should Have MQTT Messages` are dublicated.

```suggestion
    Execute Command    tedge mqtt pub te/device/main///m/test-type '{"temperature": 10}'
```

Reply:
fixed